azurerm/resource_arm_sql_database.go

```diff
@@ -118,128 +119,157 @@ func resourceArmSqlDatabase() *schema.Resource {
+	if errDb != nil {
+		d.SetId("")
+		return fmt.Errorf("Error reading SQL database %s: %v", databaseName, errDb)
 	}
 
-	resp := readResponse.Parsed.(*sql.GetDatabaseResponse)
-	d.SetId(*resp.ID)
+	d.SetId(*resultDb.ID)
 
 	return resourceArmSqlDatabaseRead(d, meta)
 }
 
 func resourceArmSqlDatabaseRead(d *schema.ResourceData, meta interface{}) error {
-	client := meta.(*ArmClient)
-	rivieraClient := client.rivieraClient
+	sqlDatabasesClient := meta.(*ArmClient).sqlDatabasesClient
 
-	readRequest := rivieraClient.NewRequestForURI(d.Id())
-	readRequest.Command = &sql.GetDatabase{}
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
 
-	readResponse, err := readRequest.Execute()
+	resGroup := id.ResourceGroup
+	serverName := id.Path["servers"]
+	databaseName := id.Path["databases"]
+
+	result, err := sqlDatabasesClient.Get(resGroup, serverName, databaseName, "")
 	if err != nil {
-		return fmt.Errorf("Error reading SQL Database: %s", err)
+		return fmt.Errorf("Error reading SQL database %s: %v", databaseName, err)
 	}
-	if !readResponse.IsSuccessful() {
-		log.Printf("[INFO] Error reading SQL Database %q - removing from state", d.Id())
+	if result.Response.StatusCode == http.StatusNotFound {
 		d.SetId("")
-		return fmt.Errorf("Error reading SQL Database: %s", readResponse.Error)
+		return nil
 	}
 
-	resp := readResponse.Parsed.(*sql.GetDatabaseResponse)
+	databaseProperties := *result.DatabaseProperties
 
-	d.Set("name", resp.Name)
-	d.Set("creation_date", resp.CreationDate)
-	d.Set("default_secondary_location", resp.DefaultSecondaryLocation)
-	d.Set("elastic_pool_name", resp.ElasticPoolName)
+	d.Set("name", databaseName)
+	d.Set("resource_group_name", resGroup)
+	d.Set("location", *result.Location)
+	d.Set("server_name", serverName)
+	d.Set("creation_date", *result.CreationDate)
+	d.Set("default_secondary_location", *result.DefaultSecondaryLocation)
 
-	flattenAndSetTags(d, resp.Tags)
+	if databaseProperties.ElasticPoolName != nil {
+		d.Set("elastic_pool_name", *databaseProperties.ElasticPoolName)
+	}
+
+	flattenAndSetTags(d, result.Tags)
 
 	return nil
 }
 
 func resourceArmSqlDatabaseDelete(d *schema.ResourceData, meta interface{}) error {
-	client := meta.(*ArmClient)
-	rivieraClient := client.rivieraClient
-
-	deleteRequest := rivieraClient.NewRequestForURI(d.Id())
-	deleteRequest.Command = &sql.DeleteDatabase{}
+	sqlDatabasesClient := meta.(*ArmClient).sqlDatabasesClient
 
-	deleteResponse, err := deleteRequest.Execute()
+	id, err := parseAzureResourceID(d.Id())
 	if err != nil {
-		return fmt.Errorf("Error deleting SQL Database: %s", err)
+		return err
 	}
-	if !deleteResponse.IsSuccessful() {
-		return fmt.Errorf("Error deleting SQL Database: %s", deleteResponse.Error)
+
+	resGroup := id.ResourceGroup
+	serverName := id.Path["servers"]
+	databaseName := id.Path["databases"]
+
+	result, error := sqlDatabasesClient.Delete(resGroup, serverName, databaseName)
+	if result.Response.StatusCode != http.StatusOK {
```

Comment:
can we update this [to check for the error and use the wrapper function as outlined in the SQL Server resource](https://github.com/terraform-providers/terraform-provider-azurerm/blob/1dbc1e7fde08c1047a2a6df4c1c33e8e142c6901/azurerm/resource_arm_sql_server.go#L157)?